lib/active_admin/orm/active_record/comments/comment.rb

```diff
@@ -4,7 +4,14 @@ class Comment < ActiveRecord::Base
     self.table_name = "#{table_name_prefix}active_admin_comments#{table_name_suffix}"
 
     belongs_to :resource, polymorphic: true
-    belongs_to :author,   polymorphic: true
+
+    # NOTE: after rails 5, :required is set to true by default and is deprecated.
+    belongs_to_optional_flag = if Rails::VERSION::MAJOR < 5
+                                 { required: false }
+                               else
+                                 { optional: true }
+                               end
+    belongs_to :author,   { polymorphic: true }.merge(belongs_to_optional_flag)
```

Comment:
Up until now, `author` was compulsory for Rails 5 aplications with `belongs_to_required_by_default` and authentication enabled. Sould we add something like

```ruby
validates :author, presence: true, if: -> { Rails.application.config.active_record.belongs_to_required_by_default && ActiveAdmin.application.authentication_method != false }
```

Reply:
I think the major problem is the `belongs_to` default value for `: required ` is different in rails 4 and 5. If we want to keep the same behavior both in rails 4 and 5, we still have to change the `:required` value, and it will auto setup a validation, adding a `validates :author` is needless. 

refs:
https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-belongs_to

```
:required
When set to true, the association will also have its presence validated. This will validate the association itself, not the id. You can use :inverse_of to avoid an extra query during validation. NOTE: required is set to true by default and is deprecated. If you don't want to have association presence validated, use optional: true.
```

Reply:
Yeah, sorry, I didn't explain it right. I mean turning off the associations validation (via `required: false` or `optional: true`, _and_ adding the presence validation that wouldn't be needless if we turn the association's validation off.

Reply:
By the way, we'll be dropping support for Rails 4 soon, so the different values here won't be a problem (although I don't fully understand why they are a problem now).

Reply:
maybe we can have a meta tag like "TODO: refactor this after dropping support for Rails 4", then we can filter these codes quickly using `rake notes`

Reply:
> Still, what's wrong with
> 
> ```ruby
> belongs_to_optional_flag = if Rails::VERSION::MAJOR < 5
>                                  { required: false }
>                                else
>                                  { optional: true }
>                                end
> belongs_to :author,   { polymorphic: true }.merge(belongs_to_optional_flag)
> 
> validates :author, presence: true, if: -> { Rails.application.config.active_record.belongs_to_required_by_default && ActiveAdmin.application.authentication_method != false }
> ```
> 
> I think it would keep the same behavior in the general case, but still fix this bug?

we should not use`validate :author, presence: true` in any case, since `:author` is optional

Reply:
Author is not currently optional for Rails 5 aplications with `belongs_to_required_by_default`. I'd like to make sure those applications still keep that behavior, but I'd like to be able to fix this bug too. Does my proposal not achieve that?

Reply:
```
belongs_to_optional_flag = if Rails::VERSION::MAJOR < 5
                                 { required: false }
                               else
                                 { optional: true }
                               end
belongs_to :author,   { polymorphic: true }.merge(belongs_to_optional_flag)
```

these codes will set `optional: true` for  Rails 5 applications, even `belongs_to_required_by_default`.

Reply:
Yes, exactly, that's why I'm suggesting to also add a presence validation that's only applied when `belongs_to_required_by_default` and authentication are enabled, in order to "restore" the validation in every case except for the one this bug applies to.

Reply:
I don't think so.
for Rails 4, `belongs_to :author` is optional, since ':require' is `false` by default.
for Rails 5, we should still keep `belongs_to :author` is optional, no matter Rails framework changed the `belongs_to_required_by_default` to whatever value.

Reply:
Ok, let's try to agree on the basics.

Say an application is currently using Rails default (`belongs_to_required_by_default = true`) and not changing that. Let's say they're also using authentication. We can assume that all the comments in their comments table have an author. Then say they run a script to bulk load some comments with a bug that tries to create some comments without an author. With the current code, the `Comment` model will raise validation errors and prevent that. If we introduce this new code and set `belongs_to :author, polymorphic: true, optional: true`, we'll change that behavior and validation errors will no longer be raised for comments without author.

Do we agree on that?

Reply:
IMO, I think this is a bugfix but not a new feature, because when using Rails 4, comment's author is optional.
But Rails 5 has changed it to required by default. "the framework has changed the application's behavior", so I'm thinking this is a bug. This PR is going to fix this issue.

Otherwise, we have to write 2 test cases, one is for Rails 4 and the other one is for Rails 5 since the behavior is changed in your written cases.

Reply:
I see your point now. According to you the expected behavior is that of Rails 4, and this fix simply restores that behavior. Specially since up until very recently we were using `belongs_to_required_by_default = false` in our specs, I understand how you may assume that.

However, I see this differently, ActiveAdmin has supported Rails 5 for a long time now, and when Rails 5 was released we didn't change this `belongs_to` relationship to be optional. That means we "accepted" the new Rails 5 feature of required `belongs_to` associations and decided to use it (but unintentionally introduced this bug). I think changing this behavior now means removing a feature that some applications could miss. This is why I suggest to keep the validation when it makes sense.

I'd like other @activeadmin/maintainers to chime in, if possible. I know it's a lenghty discussion, but I would find more opinions valuable.

Reply:
no problem, I can wait for the final conclusion. You're very professional, I'm glad to discuss and make this clear.

Reply:
ok, I will update it.